shared/health/src/test/java/io/pravega/shared/health/impl/ContributorRegistryTests.java

```diff
@@ -0,0 +1,386 @@
+               registry.validate();
+           }
+       }
+       // 3. Until no indicators are left, randomly choose an indicator to remove.
+       Collections.shuffle(contributors);
+       for (HealthContributor contributor : contributors) {
+           registry.unregister(contributor);
+           registry.validate();
+       }
+    }
+
+    private void simpleRegister() {
+        int before = registry.contributors().size();
+        registry.register(new SampleHealthyIndicator());
+        Assert.assertEquals("Expected the registration to succeed, contributor count should have increased by one.",
+                before + 1,
+                registry.contributors().size());
+    }
+
+    private void simpleUnregister() {
+        HealthContributor sample = new SampleHealthyIndicator();
+        int before = registry.contributors().size();
+        registry.unregister(sample.getName());
+        log.info("Remaining HealthContributors: {}", registry.contributors());
+        Assert.assertEquals("Expected the de-registration to succeed, contributor count should have decreased by one.",
+                before - 1,
+                registry.contributors().size());
+    }
+
+    static class TestContributorRegistry extends ContributorRegistryImpl {
+        TestContributorRegistry() {
+            super(ROOT_NAME);
+        }
+
+        // Validates that all entries in each of the internal containers point to valid references.
+        // The 'components' field is not used to determine the validity of the tree-structure.
+        public void validate() {
+            try {
+                // Each container should have the same size.
+                assert contributors.size() == children.size() && children.size() == parents.size();
+                // Validate that each container contains the same set of keys.
+                for (val entry : contributors.entrySet()) {
+                    assert children.containsKey(entry.getKey());
+                    assert parents.containsKey(entry.getKey());
+                }
+                // If above holds true, the keySet for each container should be strictly equivalent.
+                // Validate that each parent/child reference for a given contributor points to a valid contributor.
+                for (val collection : children.entrySet()) {
+                    for (val contributor : collection.getValue()) {
+                        assert contributor.equals(contributors.get(contributor.getName()));
+                    }
+                }
+                // Do the same for the parent set.
+                for (val collection : parents.entrySet()) {
+                    for (val contributor : collection.getValue()) {
+                        assert contributor.equals(contributors.get(contributor.getName()));
+                    }
+                }
+            } catch (AssertionError e) {
+                toString();
```

Comment:
What is the point of this? It will have no effect. 

Please let the exception bubble up. In general, please do not catch any `*Error`, especially `AssertionError` - those are indications that the tests found something bad.